Update website to use Docusaurus

ericnakagawa opened the pull request:
**Summary**

I have done a complete conversion of Jest's website to use Docusaurus.

Docusaurus provides all of the same features: crowdin integration, static page building, deployment to github, and algolia search.

ericnakagawa commented:
I know what the error is. I'll resolve and send an update to this this PR.

The issue is the 'yarn build' script tries building the website, but looks for localized files and fails if they are missing.

thymikee commented on .circleci/config.yml:
```diff
@@ -27,15 +27,18 @@ aliases:
       echo "machine github.com login jest-bot password $GITHUB_TOKEN" > ~/.netrc
-      # crowdin install start
+      # install Docusaurus and generate file of English strings
+      cd website && npm install && npm run write-translations && cd ..
```
Can we install the deps w yarn instead?

ericnakagawa replied:
Heck yeah! I'll swap that out now.

ericnakagawa commented:
@thymikee Hi Mike, are you able to review and accept?

thymikee commented:
I'll see what I can do!

thymikee commented on website/package.json:
```diff
@@ -1,27 +1,20 @@
+    "examples": "docusaurus-examples",
     "crowdin-upload": "crowdin-cli --config ../crowdin.yaml upload sources --auto-update -b master",
-    "crowdin-download": "crowdin-cli --config ../crowdin.yaml download -b master"
+    "crowdin-download": "crowdin-cli --config ../crowdin.yaml download -b master",
```
I don't see `crowdin-download` nor `crowdin-upload` used anywhere – is this expected and somehow used implicitly? 

Also, `README.md` would an use update (e.g. port is outdated)

thymikee commented on website/pages/en/help.js:
```diff
@@ -0,0 +1,93 @@
+
+const translate = require('../../server/translate.js').translate;
+
+// const siteConfig = require(process.cwd() + "/siteConfig.js");
```
dead code

ericnakagawa replied:
Removed.

thymikee commented on website/server/feed.js:
```diff
@@ -1,94 +0,0 @@
-/* eslint-disable sort-keys */
```
Aren't we providing RRS feed anymore?

ericnakagawa replied:
Docusaurus doesn't have an RSS feature. I've made it a TODO.

ericnakagawa replied:
New PR includes updated Docusaurus with RSS/Atom feeds.

thymikee commented on website/pages/en/help.js:
```diff
@@ -0,0 +1,93 @@
+const Container = CompLibrary.Container;
+const GridBlock = CompLibrary.GridBlock;
+
+const translate = require('../../server/translate.js').translate;
```
Is this file generated somehow? It's deleted in this PR.

ericnakagawa replied:
This is a file from Docusaurus.

thymikee replied:
Ok. I'd expect something like:
```js
const {translate} = require('docusaurus/server');
```
but not sure how this thing works currently.

thymikee commented on website/siteConfig.js:
```diff
@@ -1,39 +1,14 @@
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
```
License headers need to be adjusted to MIT

ericnakagawa replied:
I can't adjust any additional licenses at this time.

SimenB replied:
Still bsd here though

thymikee commented on website/siteConfig.js:
```diff
@@ -270,25 +245,37 @@ const users = [
+  ogImage: 'img/opengraph.png',
+  recruitingLink: 'https://crowdin.com/project/jest',
+  algolia: {
+    apiKey: '833906d7486e4059359fa58823c4ef56',
```
Should we expose Algolia API key?

ericnakagawa replied:
Good point. This would be the same issue as you not having crowdin keys to test. I'll move the keys to CI.

thymikee commented on website/static/css/custom.css:
```diff
@@ -0,0 +1,23 @@
+}
+
+@media only screen and (min-width: 1500px) {
+}
```
Unused media queries

ericnakagawa replied:
I've removed them.

orta commented:
Impressive amount of deletions here 👍

ericnakagawa commented:
@thymikee Please do another review. Hoping this update meets the expected standards! Preview -- https://build-mitsvvnzqs.now.sh/jest#

orta commented:
Not sure if this is related, but the footer is using the wrong repo for the stars:

![screen shot 2017-10-06 at 1 45 56 pm](https://user-images.githubusercontent.com/49038/31291039-c121453e-aa9c-11e7-9d25-b49a02c7e6f1.png)

thymikee commented:
@ericnakagawa I'm out for the weekend, but if I make some time, definitely have a look. Can I somehow build it locally?

ericnakagawa commented:
@orta Thanks for the catch. Latest push has matching footer info + logos.
@thymikee You can manually run by running `yarn` from `website`, but you won't be able to sync with/to crowdin since that requires having Jest's crowdin keys.

Here's a fresh build from my machine: https://build-vtlaztmazq.now.sh (currently uploading on airplane WiFi)

SimenB commented on website/.gitignore:
```diff
@@ -0,0 +1,11 @@
+node_modules
```
why a nested gitignore?

SimenB commented:
![image](https://user-images.githubusercontent.com/1404810/31371316-6f84030a-ad90-11e7-9436-3996b7cfbc9e.png)

Loving it 😀 

Only issue I found is that all the "Edit this doc" links 404 on github (they are missing the language). That might be fixed on the latest upload, it currently gives me 403

SimenB commented on website/.gitignore:
```diff
@@ -1,4 +1,3 @@
-node_modules
```
I meant the whole file, why not just stick the ignores you need in the root `.gitignore`?

ericnakagawa replied:
@SimenB Ahh, I see. In event we ever break out the website into its own repo. For time being, I'll migrate the paths to the top level `.gitignore`.

ericnakagawa commented:
@SimenB I removed the line from .gitignore and updated the edit doc path to include `/en`.

As for the sidebar links going to `#content` I've added this as an issue on the Docusaurus repo -- I'll review and fix there, then update in Jest once we decide. https://github.com/facebookexperimental/Docusaurus/issues/120

ericnakagawa commented:
Due to my current internet connection I am unable to troubleshoot the breaking CI test.

SimenB commented:
The CI failure is just prettier it seems like. So a `yarn lint --fix` should make everything nice and happy

ericnakagawa commented:
@thymikee @SimenB Requesting a review! I had to manually re-run a job because of a segfault.

SimenB commented:
Only thing I could spot was the comment about the license, nothing else I noticed (either in the code or clicking through https://build-qbmkjpokcg.now.sh/jest)

ericnakagawa commented:
@SimenB Updated last file, also updated package.json to use latest docusaurus with MIT license throughout.

SimenB commented on website/siteConfig.js:
```diff
@@ -3,279 +3,278 @@
-    image: '/jest/img/logos/facebook.png',
-    infoLink: 'https://code.facebook.com',
-    pinned: true,
+    caption: "Facebook",
```
your autoformat kicked in, I think

ericnakagawa replied:
🤦‍♂️

cpojer replied:
lol

ericnakagawa commented:
Rebased and waiting for tests.

SimenB commented:
Seeing as this PR moves all files out of `en` directory, I'm starting to wonder if https://github.com/facebook/jest/pull/4536/commits/1fc503bd23eeddb38969dd050974d178deaec31c#diff-c272134cac3612b41f93893bbaa34d24R254 was a mistake?

ericnakagawa commented:
Rebased again after docs updated.

ericnakagawa commented:
@SimenB Ah, I knew I had a reason for configuring it this way. I've rolled back the change.

Rebased, and ran through lint fixer.

We should be good to go!
